packages/web-components/src/switch/switch.styles.ts

```diff
@@ -142,4 +146,54 @@ export const styles = css`
   :host([aria-checked='true'][disabled]) .checked-indicator {
     background: ${colorNeutralForegroundDisabled};
   }
-`;
+
+  :host(:focus-visible) {
+    border-color: ${colorTransparentStroke};
+    outline: ${strokeWidthThick} solid ${colorTransparentStroke};
+    box-shadow: ${shadow4}, 0 0 0 2px ${colorStrokeFocus2};
+  }
+`.withBehaviors(
+  forcedColorsStylesheetBehavior(css`
+    .switch {
+      border-color: InactiveBorder;
+    }
+    :host(:hover) .switch {
+      border-color: InactiveBorder;
+    }
+    :host(:active) .switch {
+      border-color: ActiveBorder;
+    }
+    :host([aria-checked='true']) .switch {
+      background: Highlight;
+      border-color: Highlight;
+    }
+    :host([aria-checked='true']:hover) .switch {
+      background: Highlight;
+      border-color: Highlight;
+    }
+    :host([aria-checked='true']:active) .switch {
+      background: Highlight;
+      border-color: Highlight;
+    }
+
+    .checked-indicator {
+      background-color: ActiveCaption;
+    }
+    :host(:hover) .checked-indicator {
+      background-color: ActiveCaption;
+    }
+    :host(:active) .checked-indicator {
+      background-color: ActiveCaption;
+    }
```

Comment:
Likewise can these also be a single selector? Let's look for duplication like this elsewhere and fix as well please :)